projects/storefrontlib/src/cms-components/asm/customer-selection/customer-selection.component.ts

```diff
@@ -47,23 +54,27 @@ export class CustomerSelectionComponent implements OnInit, OnDestroy {
       this.form.controls.searchTerm.valueChanges
         .pipe(debounceTime(300))
         .subscribe(value => {
-          if (!!this.selectedCustomer && value !== this.selectedCustomer.name) {
-            this.selectedCustomer = undefined;
-          }
-          if (!!this.selectedCustomer) {
-            return;
-          }
-          this.asmService.customerSearchReset();
-          if (value.trim().length >= 3) {
-            this.asmService.customerSearch({
-              query: value,
-              pageSize: this.config.asm.customeSearch.maxResults,
-            });
-          }
+          this.handleSearchTerm(value);
         })
     );
   }
 
+  private handleSearchTerm(value: string) {
+    if (!!this.selectedCustomer && value !== this.selectedCustomer.name) {
+      this.selectedCustomer = undefined;
+    }
+    if (!!this.selectedCustomer) {
```

Comment:
switch to Boolean constructor instead of !! for readability

Reply:
Fixed